tests/generic_config_updater/test_kubernetes_config.py

```diff
@@ -0,0 +1,266 @@
+    },
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER/SERVER/disable"
+    }
+]
+K8SHALFTOEMPTYPATCH = [
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER"
+    }
+]
+K8SEMPTYTOFULLPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER",
+        "value": {"SERVER": {"disable": "false", "ip": "k8svip.ap.gbl"}}
+    }
+]
+K8SFULLTOFULLPATCH = [
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": "k8svip2.ap.gbl"
+    }
+]
+K8SFULLTOWRONGIPPATCH = [
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": ""
+    },
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/disable",
+        "value": "true"
+    }
+]
+K8SFULLTOEMPTYPATCH = [
+    {
+        "op": "remove",
+        "path": "/KUBERNETES_MASTER"
+    }
+]
+K8SEMPTYTOWRONGIPPATCH = [
+    {
+        "op": "add",
+        "path": "/KUBERNETES_MASTER",
+        "value": {"SERVER": {"disable": "true", "ip": ""}}
+    }
+]
+K8SWRONGIPTOFULLPATCH = [
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/ip",
+        "value": "k8svip.ap.gbl"
+    },
+    {
+        "op": "replace",
+        "path": "/KUBERNETES_MASTER/SERVER/disable",
+        "value": "false"
+    }
+]
+
+
+# Succeed and fail flag
+SUCCEED = "SUCCEED"
+FAIL = "FAIL"
+
+
+test_data_1 = {
+    0: (K8SEMPTYTOHALFPATCH, K8SHALFCONFIG, SUCCEED),
+    1: (K8SHALFTOWRONGIPPATCH, K8SHALFCONFIG, FAIL),
+    2: (K8SHALFTOFULLPATCH, K8SFULLCONFIG, SUCCEED),
+    3: (K8SFULLTOHALFPATCH, K8SHALFCONFIG, SUCCEED),
+    4: (K8SHALFTOEMPTYPATCH, K8SEMPTYCONFIG, SUCCEED),
+    5: (K8SEMPTYTOFULLPATCH, K8SFULLCONFIG, SUCCEED),
+    6: (K8SFULLTOFULLPATCH, K8SFULLCONFIG2, SUCCEED),
+    7: (K8SFULLTOWRONGIPPATCH, K8SFULLCONFIG2, FAIL),
+    8: (K8SFULLTOEMPTYPATCH, K8SEMPTYCONFIG, SUCCEED),
+    9: (K8SEMPTYTOWRONGIPPATCH, K8SEMPTYCONFIG, FAIL),
+}
+
+
+test_data_2 = {
+    10: (K8SWRONGIPTOFULLPATCH, K8SEMPTYIPCONFIG, FAIL),
+}
+
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for k8s config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    original_k8s_config = get_k8s_runningconfig(duthost)
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+        current_k8s_config = get_k8s_runningconfig(duthost)
+        pytest_assert(
+            set(original_k8s_config) == set(current_k8s_config),
+            "k8s config are not suppose to change after test org: {}, cur: {}"
+            .format(original_k8s_config, current_k8s_config)
+        )
+
+    finally:
+        delete_checkpoint(duthost)
+
+
+def get_k8s_runningconfig(duthost):
```

Comment:
Any other check method other than running config?
The success of  apply-patch operation should guarantee the CONFIG_DB update. 
Is there a show command such as  `show k8s server`?

Reply:
Thanks for this comment, have add 'show kube server' related validation.